cmake/compiler/gcc/target.cmake

```diff
@@ -5,7 +5,14 @@ set_ifndef(C++ g++)
 # Configures CMake for using GCC, this script is re-used by several
 # GCC-based toolchains
 
-find_program(CMAKE_C_COMPILER ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+if("${SPARSE}" STREQUAL "y")
+  find_program(CMAKE_C_COMPILER cgcc)
+  find_program(REAL_CC ${CROSS_COMPILE}${CC} PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+  set(ENV{REAL_CC} ${REAL_CC})
```

Comment:
this will not work as you expect.

It only set the environment variable for this CMake session, but not when invoking the build tool (such as `ninja`).
The only reason it might have worked for you is if you have had this set in the environment already or it has picked up a default compiler in your system that happened to work when you were testing.

> `set(ENV{<variable>} [<value>])`
> 
> Sets an [Environment Variable](https://cmake.org/cmake/help/latest/manual/cmake-env-variables.7.html#manual:cmake-env-variables(7)) to the given value. Subsequent calls of $ENV{<variable>} will return this new value.
>
> This command affects only the current CMake process, not the process from which CMake was called, nor the system environment at large, nor the environment of subsequent build or test processes.
https://cmake.org/cmake/help/latest/command/set.html#set-environment-variable

also see here: https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#how-can-i-get-or-set-environment-variables

and from what I can tell, `cgcc` requires `REAL_CC` to be set in environment, else it will default to something else.
https://www.man7.org/linux/man-pages/man1/cgcc.1.html
> REAL_CC
> - If set, cgcc will use this as the compiler to invoke, rather than the default cc.

Reply:
> this will not work as you expect.

Addressed in follow-up PR
- #50398